lib/interceptor/cache.js

```diff
@@ -168,6 +173,13 @@ export class CacheStore {
     rows.map((i) => {
       i.data = JSON.parse(i.data)
       i.vary = JSON.parse(i.vary)
+      i.data = {
+        ...i.data,
+        // JSON.parse doesn't convert a Buffer object back to a Buffer object once it has been stringified.
+        body: this.#convertToBuffer(i.data.body),
+        rawHeaders: this.#convertToBuffer(i.data.rawHeaders),
+        rawTrailers: this.#convertToBuffer(i.data.rawTrailers),
```

Comment:
Why convert to buffer? I think it's better if you save headers and trailers as strings.